tests/jlm/hls/backend/rvsdg2rhls/MemoryConverterTests.cpp

```diff
@@ -147,7 +149,75 @@ TestLoad()
 
   return 0;
 }
-JLM_UNIT_TEST_REGISTER("jlm/hls/backend/rvsdg2rhls/MemoryConverterTests-2", TestLoad)
+JLM_UNIT_TEST_REGISTER("jlm/hls/backend/rvsdg2rhls/MemoryConverterTests-Load", TestLoad)
+
+static int
+TestStore()
+{
+  using namespace jlm::llvm;
+  using namespace jlm::hls;
+
+  auto rvsdgModule = RvsdgModule::Create(jlm::util::filepath(""), "", "");
+
+  // Setup the function
+  std::cout << "Function Setup" << std::endl;
+  auto functionType = jlm::rvsdg::FunctionType::Create(
+      { jlm::llvm::PointerType::Create(),
+        jlm::rvsdg::bittype::Create(32),
+        MemoryStateType::Create() },
+      { MemoryStateType::Create() });
+
+  auto lambda = jlm::rvsdg::LambdaNode::Create(
+      rvsdgModule->Rvsdg().GetRootRegion(),
+      LlvmLambdaOperation::Create(functionType, "test", linkage::external_linkage));
+
+  // Single load
+  auto storeAddress = lambda->GetFunctionArguments()[0];
+  auto storeData = lambda->GetFunctionArguments()[1];
+  auto memoryStateArgument = lambda->GetFunctionArguments()[2];
+  auto storeOutput =
+      StoreNonVolatileNode::Create(storeAddress, storeData, { memoryStateArgument }, 32);
+
+  auto lambdaOutput = lambda->finalize({ storeOutput[0] });
+  jlm::llvm::GraphExport::Create(*lambdaOutput, "f");
+
+  // Act
+  jlm::rvsdg::view(rvsdgModule->Rvsdg(), stdout);
+  MemoryConverter(*rvsdgModule);
+  jlm::rvsdg::view(rvsdgModule->Rvsdg(), stdout);
+
+  // Memory Converter replaces the lambda so we start from the root of the graph
+  auto region = &rvsdgModule->Rvsdg().GetRootRegion();
+  assert(region->nnodes() == 1);
+  lambda = jlm::util::AssertedCast<jlm::rvsdg::LambdaNode>(region->Nodes().begin().ptr());
+
+  // Assert
+  auto lambdaRegion = lambda->subregion();
+  assert(lambdaRegion->nnodes() == 2);
+  assert(lambdaRegion->narguments() == 4);
+  assert(lambdaRegion->nresults() == 2);
+
+  // Memory state
+  jlm::util::AssertedCast<const MemoryStateType>(&lambdaRegion->result(0)->origin()->type());
+
+  // Store
+  auto storeNode =
+      jlm::util::AssertedCast<jlm::rvsdg::node_output>(lambdaRegion->result(0)->origin())->node();
+  jlm::util::AssertedCast<const store_op>(&storeNode->GetOperation());
+
+  // Request Node
+  auto requestNode =
+      jlm::util::AssertedCast<jlm::rvsdg::node_output>(lambdaRegion->result(1)->origin())->node();
+  jlm::util::AssertedCast<const mem_req_op>(&requestNode->GetOperation());
+
+  // Request source
+  auto requestSource = requestNode->input(0)->origin();
+  storeNode = jlm::util::AssertedCast<jlm::rvsdg::node_output>(requestSource)->node();
+  jlm::util::AssertedCast<const store_op>(&storeNode->GetOperation());
+
```

Comment:
Use normal asserts for assertions in unit tests.

Reply:
Changed